### PR TITLE
ADCM-3060 add ui test for filter on users page

### PR DESCRIPTION
**Pull request by annademina**

### CODE REVIEW

File: tests/ui_tests/app/page/admin/page.py [hunk 1]
```diff
@@ -359,6 +359,29 @@ def check_delete_button_not_presented(self, username: str):
...
+        self.find_and_click(AdminUsersLocators.filter_btn)
+        self.wait_element_visible(AdminUsersLocators.FilterPopup.block)
+        for filter_item in self.find_elements(AdminUsersLocators.FilterPopup.filter_item):
+            if filter_item.text.lower() == filter_name.lower():
+                filter_item.click()
+        self.wait_element_visible(AdminUsersLocators.filter_dropdown_select).click()
+        self.wait_element_visible(AdminUsersLocators.filter_dropdown_option)
+        for filter_option in self.find_elements(AdminUsersLocators.filter_dropdown_option):
+            if filter_option.text.lower() == filter_option_name.lower():
+                filter_option.click()
```
Comment (Sealwing) [inline tests/ui_tests/app/page/admin/page.py:375]: What if it's not found?

File: tests/ui_tests/test_admin_page.py [hunk 2]
```diff
@@ -397,13 +397,41 @@ def test_add_ldap_group_to_users(self, user, users_page, sdk_client_fs, ldap_use
...
             user_row = users_page.get_user_row_by_username(ldap_user_in_group['name'])
             assert test_group.name in user_row.text, "User group didn't changed"
 
+    @pytest.mark.ldap()
+    @pytest.mark.usefixtures("configure_adcm_ldap_ad")
+    def test_filter_users(self, user, users_page, sdk_client_fs, ldap_user_in_group):
+        """Check that users can be filtered"""
+
+        with allure.step("Wait ldap integration ends"):
+            users_page.header.wait_success_job_amount_from_header(1)
```
Comment (Sealwing) [inline tests/ui_tests/test_admin_page.py:414]: Maybe just launch sync action?
It seems faster than to wait ~1min

File: tests/ui_tests/test_admin_page.py [hunk 3]
```diff
@@ -397,13 +397,41 @@ def test_add_ldap_group_to_users(self, user, users_page, sdk_client_fs, ldap_use
...
+        users_page.driver.refresh()
+        users_page.filter_users_by("status", "active")
+        with allure.step("Check users are filtered by active status"):
+            assert len(users_page.get_all_user_rows()) == 4, "Should be 4 visible active users"
```
Comment (Sealwing) [inline tests/ui_tests/test_admin_page.py:418]: I think we need to check if all those users are actually active (by name, e.g.)
Or at least check that "active state" is "active" in all rows

File: tests/ui_tests/test_admin_page.py [hunk 4]
```diff
@@ -397,13 +397,41 @@ def test_add_ldap_group_to_users(self, user, users_page, sdk_client_fs, ldap_use
...
+        users_page.filter_users_by("type", "local")
+        with allure.step("Check users are filtered by local type"):
+            assert len(users_page.get_all_user_rows()) == 3, "Should be 0 visible local users"
+        users_page.remove_user_filter()
+        users_page.filter_users_by("type", "ldap")
+        with allure.step("Check users are filtered by ldap status"):
+            assert len(users_page.get_all_user_rows()) == 1, "Should be 0 visible ldap users"
+        users_page.filter_users_by("status", "active")
+        with allure.step("Check users are filtered both by active status and ldap"):
+            assert len(users_page.get_all_user_rows()) == 1, "Should be 0 visible active ldap users"
```
Comment (Sealwing) [inline tests/ui_tests/test_admin_page.py:433]: 1. You check for one value and another value's in message
2. Same thing with checks, as I've said above

**Review by Sealwing — CHANGES_REQUESTED** [review]

File: tests/ui_tests/app/page/admin/page.py (hunk 1 above)
Comment (annademina) [reply tests/ui_tests/app/page/admin/page.py]: changed

File: tests/ui_tests/test_admin_page.py (hunk 3 above)
Comment (annademina) [reply tests/ui_tests/test_admin_page.py]: changed

File: tests/ui_tests/test_admin_page.py (hunk 4 above)
Comment (annademina) [reply tests/ui_tests/test_admin_page.py]: yep, changed

File: tests/ui_tests/test_admin_page.py (hunk 2 above)
Comment (annademina) [reply tests/ui_tests/test_admin_page.py:414]: tried wait_for_task_and_assert_result(sdk_client_fs.adcm().action(name="run_ldap_sync").task(), 'success') but it always fails with ObjectNotFound

Comment (annademina) [reply tests/ui_tests/test_admin_page.py]: +

File: tests/ui_tests/app/page/admin/page.py (hunk 1 above)
Comment (Sealwing) [reply tests/ui_tests/app/page/admin/page.py]: It's the same, as I see :/

File: tests/ui_tests/test_admin_page.py [hunk 5]
```diff
@@ -389,21 +390,64 @@ def test_add_user_to_group(self, user, users_page, sdk_client_fs):
...
+    @pytest.mark.usefixtures("configure_adcm_ldap_ad")
+    def test_filter_users(self, user, users_page, sdk_client_fs, ldap_user_in_group):
+        """Check that users can be filtered"""
+
+        with allure.step("Wait ldap integration ends"):
+            wait_for_task_and_assert_result(sdk_client_fs.adcm().action(name="run_ldap_sync").run(), 'success')
+        users_page.driver.refresh()
+        users_page.filter_users_by("status", "active")
+        with allure.step("Check users are filtered by active status"):
+            active_users = sdk_client_fs.user_list(is_active=True).data
```
Comment (Sealwing) [inline tests/ui_tests/test_admin_page.py:419]: no need to call `.data`

File: tests/ui_tests/test_admin_page.py [hunk 6]
```diff
@@ -389,21 +390,64 @@ def test_add_user_to_group(self, user, users_page, sdk_client_fs):
...
+            assert len(users_page.get_all_user_rows()) == len(
+                active_users
+            ), f"Should be {len(active_users)} visible active users"
+        users_page.remove_user_filter()
+        users_page.filter_users_by("status", "inactive")
+        with allure.step("Check users are filtered by inactive status"):
+            inactive_users = sdk_client_fs.user_list(is_active=False).data
```
Comment (Sealwing) [inline tests/ui_tests/test_admin_page.py:426]: same "no need to call .data"

File: tests/ui_tests/test_admin_page.py [hunk 7]
```diff
@@ -389,21 +390,64 @@ def test_add_user_to_group(self, user, users_page, sdk_client_fs):
...
+            ldap_users = sdk_client_fs.user_list(type='ldap').data
+            assert len(users_page.get_all_user_rows()) == len(
+                ldap_users
+            ), f"Should be {len(ldap_users)} visible ldap users"
+        users_page.filter_users_by("status", "active")
+        with allure.step("Check users are filtered both by active status and ldap"):
+            active_ldap_users = sdk_client_fs.user_list(is_active=True, type='ldap').data
+            assert len(users_page.get_all_user_rows()) == len(
+                active_ldap_users
+            ), f"Should be {active_ldap_users} visible active ldap users"
```
Comment (Sealwing) [inline tests/ui_tests/test_admin_page.py:449]: here and above lengths can be the same when the values differ
I suggest using something like `all(lambda u: u.username in usernames_from_ui, sdk_client_fs.user_list(is_active=True, type='ldap'))`
where `usernames_from_ui` will be usernames from UI table

**Review by Sealwing — CHANGES_REQUESTED** [review]

File: tests/ui_tests/test_admin_page.py (hunk 3 above)
Comment (Sealwing) [reply tests/ui_tests/test_admin_page.py]: commented below

File: tests/ui_tests/test_admin_page.py (hunk 5 above)
Comment (annademina) [reply tests/ui_tests/test_admin_page.py]: removed

File: tests/ui_tests/test_admin_page.py (hunk 6 above)
Comment (annademina) [reply tests/ui_tests/test_admin_page.py]: removed

File: tests/ui_tests/test_admin_page.py (hunk 7 above)
Comment (annademina) [reply tests/ui_tests/test_admin_page.py]: check by names)

**Review by Sealwing — APPROVED** [review]

**Pull request merged**